Makefile

```diff
@@ -318,10 +337,19 @@ docker-images-elastic:
 	docker build -t $(DOCKER_NAMESPACE)/jaeger-es-rollover:${DOCKER_TAG} plugin/storage/es -f plugin/storage/es/Dockerfile.rollover
 	@echo "Finished building jaeger-es-indices-clean =============="
 
-.PHONY: docker-images-jaeger-backend
-docker-images-jaeger-backend:
+docker-images-jaeger-backend: TARGET = release
+docker-images-jaeger-backend-debug: TARGET = debug
+docker-images-jaeger-backend-debug: SUFFIX = -debug
+
+.PHONY: docker-images-jaeger-backend docker-images-jaeger-backend-debug
+docker-images-jaeger-backend docker-images-jaeger-backend-debug: create-baseimg create-debugimg
 	for component in agent collector query ingester ; do \
-		docker build -t $(DOCKER_NAMESPACE)/jaeger-$$component:${DOCKER_TAG} cmd/$$component --build-arg TARGETARCH=$(GOARCH) ; \
+		docker build --target $(TARGET) \
+			--tag $(DOCKER_NAMESPACE)/jaeger-$$component:${DOCKER_TAG}$(SUFFIX) \
```

Comment:
This will yield a tag like this: `jaegertracing/jaeger-agent:latest-debug`. I think I would prefer `jaegertracing/jaeger-agent-debug:latest` instead, but I'm open to hear arguments/experiences with debug versions as well.